### PR TITLE
Davical 1.1.7

### PR DESCRIPTION
**Pull request by tillschaefer**
closes: https://bugs.gentoo.org/573218

- updated header
- updated EAPI to 6
- removed patches applied upstream
- adjust to new folder structure of davical tarball

### CODE REVIEW

File: dev-php/awl/awl-0.58.ebuild [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+KEYWORDS="~amd64 ~x86"
+IUSE="doc test"
+
+DEPEND="test? ( dev-php/phpunit )"
+RDEPEND="dev-lang/php:*[pdo,postgres,xml]"
+
+src_unpack() {
+	default_src_unpack
+	mv awl-* awl-${PV} || die "tarballs root folder not found"
+}
```
Comment (SoapGentoo) [inline dev-php/awl/awl-0.58.ebuild]: scrap all of this and just inherit `vcs-snapshot`

File: dev-php/awl/awl-0.58.ebuild [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	:
+}
+
+src_test() {
+	phpunit tests/ || die "test suite failed"
+}
+
+src_install() {
+	dodoc debian/changelog
+	use doc && dohtml -r "docs/api/"
```
Comment (SoapGentoo) [inline dev-php/awl/awl-0.58.ebuild]: dohtml is deprecated in EAPI 6. Do
```
HTML_DOCS=( docs/api/. )
einstalldocs
```

File: dev-php/awl/awl-0.58.ebuild [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+	insinto "/usr/share/php/${PN}"
```
Comment (SoapGentoo) [inline dev-php/awl/awl-0.58.ebuild]: drop the quotes, not needed, `${PN}` will never contain a space

File: www-apps/davical/davical-1.1.6.ebuild [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+	dev-perl/YAML
+	>=dev-php/awl-0.58
+	virtual/httpd-php"
+
+need_httpd
+
+src_unpack() {
+	default_src_unpack
+	mv davical-* ${P} || die "tarballs root folder not found"
+}
```
Comment (SoapGentoo) [inline www-apps/davical/davical-1.1.6.ebuild]: same here

File: www-apps/davical/davical-1.1.6.ebuild [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+src_install() {
+	webapp_src_preinst
+
+	dodoc INSTALL README debian/README.Debian \
+		testing/README.regression_tests TODO debian/changelog
+
+	einfo "Installing web files"
+	insinto "${MY_HTDOCSDIR}"
+	doins -r htdocs/* htdocs/.htaccess
```
Comment (SoapGentoo) [inline www-apps/davical/davical-1.1.6.ebuild]: `doins -r htdocs/. htdocs/.htaccess`

File: www-apps/davical/davical-1.1.6.ebuild [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+	einfo "Installing main files and i18n"
+	insinto "${MY_HOSTROOTDIR}/${PN}"
+	doins -r inc locale
+	rm "${D}/${MY_HOSTROOTDIR}/${PN}/inc/always.php.in" || die
```
Comment (SoapGentoo) [inline www-apps/davical/davical-1.1.6.ebuild]: `rm "${ED}/${MY_HOSTROOTDIR}/${PN}/inc/always.php.in" || die`

File: www-apps/davical/davical-1.1.6.ebuild [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+	einfo "Installing sql files"
+	insinto "${MY_SQLSCRIPTSDIR}"
+	doins -r dba/*
```
Comment (SoapGentoo) [inline www-apps/davical/davical-1.1.6.ebuild]: `doins -r dba/.` (avoid unnecessary globs)

File: www-apps/davical/davical-1.1.6.ebuild [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+
+	if use doc ; then
+		einfo "Installing documentation"
+		dohtml -r docs/api/ docs/website/
```
Comment (SoapGentoo) [inline www-apps/davical/davical-1.1.6.ebuild]: `dohtml` again

File: www-apps/davical/davical-1.1.6.ebuild [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+	fi
+
+	insinto /etc/${PN}
+	doins config/* "${FILESDIR}/vhost-example"
```
Comment (SoapGentoo) [inline www-apps/davical/davical-1.1.6.ebuild]: globs

File: www-apps/davical/davical-1.1.6.ebuild [hunk 10]
```diff
@@ -0,0 +1,69 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit webapp eutils
```
Comment (SoapGentoo) [inline www-apps/davical/davical-1.1.6.ebuild]: why do you inherit `eutils`?

**Comment by tillschaefer** [comment]
updated to respect change requests
remove doc useflag since doc generation is broken

File: dev-php/awl/awl-0.58.ebuild [hunk 11]
```diff
@@ -0,0 +1,31 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+inherit vcs-snapshot eutils
```
Comment (SoapGentoo) [inline dev-php/awl/awl-0.58.ebuild]: why is `eutils` still here?

Comment (tillschaefer) [reply dev-php/awl/awl-0.58.ebuild]: leftover from einstalldocs .. will remove it

**Comment by SoapGentoo** [comment]
@grknight @orlitzky please review, I am happy now

**Comment by tillschaefer** [comment]
bumped to new version which was just released upstream (identical ebuild)

**Comment by orlitzky** [comment]
The latest changelog entry suggests that the doc generation might be unbroken now:

```
2018-01-11 Florian Schlichting <fsfs@debian.org>
 	* release awl 0.59
 	* switch to doxygen for api docs
```

Up to you if you feel like trying to get it to work.

**Comment by tillschaefer** [comment]
If thats ok, I would like to leave it that way for now (without doc).

**Comment by orlitzky** [comment]
No problem, we can add it later if anyone cares.

Does the awl dependency in the davical ebuild need to be updated now?

**Comment by tillschaefer** [comment]
yes, you are right.

**Comment by orlitzky** [comment]
I pulled the awl commit a second ago, thanks. For davical... does this work?

```
doins config/. "${FILESDIR}/vhost-example"
```

I got no config files installed, and it looks like there should be an `-r` there.

Once that's fixed, I don't know how the webapp-config stuff works, so you might want to get patrick to ACK that on IRC, or just go ahead with a promise that you'll fix it if it doesn't work for some reason.

**Comment by tillschaefer** [comment]
Regarding the doins: This is changed now. I did not noticed the problem, since the missing files are only example configurations.

Since I cannot reach Patrick on IRC, I am willing to respond to any non-working bugs that are coming up. Should I add myself as proxy-maintainer or how can I get notified about new bugs otherwise?

**Comment by orlitzky** [comment]
I pinged patrick on IRC (he goes by **bonsaikitten** at the moment).

**Comment by orlitzky** [comment]
> <bonsaikitten> mjo: I'm not currently using it, so feel free to take over

So feel free to update metadata.xml.

**Comment by tillschaefer** [comment]
take over = remove patrick or just add me?

**Comment by orlitzky** [comment]
Yeah that's fine. If he's not using it anymore, he can do without the bug mail. @SoapGentoo I guess you'll want to create him a maintainer bug and all that stuff, but otherwise this LGTM now.

**Comment by tillschaefer** [comment]
I already have a maintainer bug: https://bugs.gentoo.org/643946

**Pull request closed without merging**